jepsen/src/jepsen/control.clj

```diff
@@ -121,17 +121,16 @@
 
 (defn throw-on-nonzero-exit
   "Throws when the result of an SSH result has nonzero exit status."
-  [result]
-  (if (zero? (:exit result))
+  [{:keys [exit action host out err in] :as result}]
+  (if (zero? exit)
     result
-    (throw
-      (RuntimeException.
-        (str (:cmd (:action result))
-             " returned non-zero exit status " (:exit result)
-             " on " (:host result) ". STDOUT:\n"
-             (:out result)
-             "\n\nSTDERR:\n"
-             (:err result))))))
+    (throw+
+     (let [cmd (:cmd action)
+           message (str cmd " returned non-zero exit status " exit " on " host
+                        ". STDOUT:\n" out "\n\nSTDERR:\n" err)]
+       (merge {:throwable (RuntimeException. message)
```

Comment:
Yeah we're throwing a map here, which is subject to all of the slingshot destructuring. I included the original exceptions in the :throwable key just in case there were tests relying on them. I tested it out though, and catching RuntimeException here isn't backwards compatible -- so I misunderstood part of the docs. I can remove the RuntimeException from the map w/o much fuss.

Reply:
Cool cool, thanks Kit!